webapp/tests/elster_client/mock_erica.py

```diff
@@ -345,3 +356,29 @@ def errors_from_error_flags(show_response):
         # InvalidTaxNumberError
         if MockErica.invalid_tax_number_error_occurred:
             return get_json_response('invalid_tax_number')
+
+    request_id_count = {}
+
+    @staticmethod
+    def post_dummy_job(*args, **kwargs):
+        if Config.MOCK_DELIVER_FAIL_ON_POST:
+            return {"errorCode": -1, "errorMessage": "Job could not be submitted."}, 422
+        else:
+            request_id = str(uuid.uuid4())
+            MockErica.request_id_count.setdefault(request_id, 0)
+            return "/02/get_job/" + request_id, 201
+
+    @staticmethod
+    def get_dummy_job(request_id):
+        count = MockErica.request_id_count.get(request_id, -1) + 1
```

Comment:
Just fyi because I was thinking about it: As this is dummy code, I think that this is fine. However, in other cases I would think that we should catch a missing request_id more explicit:
```python
count = MockErica.request_id_count.get(request_id, None) + 1

if count is None:
    response = {"errorCode": -1, "errorMessage": "Request ID not found"}, 500

#....
```

Otherwise we could get errors if the error count is set to zero at some point and we assume the request id is missing

Reply:
I just followed the tech spec. There you only defined a 422 for errors, but yes we should use 404 for missing request id. So 422 is ok here? And also the way is handled with elif count == 0 ?